Neos.ContentRepository/Classes/Domain/Model/ExpressionBasedNodeLabelGenerator.php

```diff
@@ -66,14 +66,17 @@ public function initializeObject()
     }
 
     /**
-     * Render a node label
+     * Render a node label based on an eel expression or return the expression if it is not valid eel.
      *
      * @param \Neos\ContentRepository\Domain\Projection\Content\NodeInterface $node
      * @return string
      * @throws \Neos\Eel\Exception
      */
     public function getLabel(\Neos\ContentRepository\Domain\Projection\Content\NodeInterface $node): string
     {
+        if (Utility::parseEelExpression($this->getExpression()) === null) {
```

Comment:
While parseEelExpression is the correct way, the Regel is quite expensive (and two lines below the same refer will be rerun internally)

i don’t know if a simple str_starts with ${ would be better … it’s faster

Reply:
I know, but as you say, it's anyway being executed and I don't want to rely on the exception it throws.
And just relying on `${` is also messy ...

Reply:
yes btw i optimized this Utility::parseEelExpression in https://github.com/neos/flow-development-collection/pull/2979 so it wont matter in the future ^^